src/ripple/app/ledger/Ledger.cpp

```diff
@@ -225,14 +225,14 @@ Ledger::Ledger (uint256 const& parentHash,
     loaded = true;
 
     if (info_.txHash.isNonZero () &&
-        !txMap_->fetchRoot (info_.txHash, nullptr))
+        !txMap_->fetchRoot (SHAMapHash{info_.txHash}, nullptr))
```

Comment:
Just thinking out loud.  There are 20 of these places where you have to construct a local `SHAMapHash`.  Six of them are to pass a parameter to `fetchRoot()`.  Five are to pass a parameter to `addRootNode()`.  Would it be worthwhile making overloads of (just) those two methods to accept a `uint256`?  Or would that be taking a step backwards?  It looks like the cost of these local `SHAMapHash`s is a 256-bit `memcpy`...